### ISSUE TITLE
Error thrown: "undefined method `resbody_type?' for :init:Symbol"

### ISSUE THREAD

**Issue by jimm** [rubocop/rubocop#11084]
Parallel assignment of class vars in the input Ruby file being checked causes Rubocop to throw an error. Below is a simple Ruby file that causes Rubocop to throw an exception.

--------

## Expected behavior

The problems with the file are reported an no error is thrown. Rubocop does seem to successfully report problems with the Ruby file, however, after the error trace is output.

## Actual behavior

The input Ruby file is below. Here is the output produced when running `rubocop -d /tmp/foo.rb`.

```
For /private/tmp: Default configuration from /Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /tmp/foo.rb
An error occurred while Style/ParallelAssignment cop was inspecting /tmp/foo.rb:5:4.
undefined method `resbody_type?' for :init:Symbol

        node&.resbody_type? && @modifier_locations.include?(node.loc.keyword)
            ^^^^^^^^^^^^^^^
Did you mean?  respond_to?
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/mixin/rescue_node.rb:14:in `rescue_modifier?'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/style/parallel_assignment.rb:84:in `assignment_corrector'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/style/parallel_assignment.rb:49:in `autocorrect'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/style/parallel_assignment.rb:39:in `block in on_masgn'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/base.rb:346:in `correct'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/base.rb:127:in `add_offense'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/style/parallel_assignment.rb:39:in `on_masgn'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `public_send'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:100:in `block (2 levels) in trigger_responding_cops'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:99:in `block in trigger_responding_cops'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `each'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:98:in `trigger_responding_cops'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:69:in `on_masgn'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:160:in `on_defs'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_defs'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:153:in `on_class'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-ast-1.21.0/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cop/team.rb:83:in `investigate'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:315:in `inspect_file'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:259:in `block in do_inspection_loop'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:293:in `block in iterate_until_no_changes'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `loop'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:286:in `iterate_until_no_changes'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:255:in `do_inspection_loop'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:138:in `block in file_offenses'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:163:in `file_offense_cache'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:137:in `file_offenses'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:128:in `process_file'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:109:in `block in each_inspected_file'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `reduce'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:108:in `each_inspected_file'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:94:in `inspect_files'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/runner.rb:47:in `run'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli.rb:72:in `run_command'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli.rb:79:in `execute_runners'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/lib/rubocop/cli.rb:48:in `run'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/exe/rubocop:19:in `block in <top (required)>'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/jimm/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/rubocop-1.36.0/exe/rubocop:19:in `<top (required)>'
/Users/jimm/.rbenv/versions/3.1.2/bin/rubocop:25:in `load'
/Users/jimm/.rbenv/versions/3.1.2/bin/rubocop:25:in `<main>'
C

Offenses:

/tmp/foo.rb:1:1: C: [Correctable] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
#!/usr/bin/env ruby
^
/tmp/foo.rb:3:1: C: Style/Documentation: Missing top-level documentation comment for class Foo.
class Foo
^^^^^^^^^
/tmp/foo.rb:5:5: C: Style/ClassVars: Replace class var @@cvar_one with a class instance var.
    @@cvar_one, @@cvar_two = 1, 2
    ^^^^^^^^^^
/tmp/foo.rb:5:17: C: Style/ClassVars: Replace class var @@cvar_two with a class instance var.
    @@cvar_one, @@cvar_two = 1, 2
                ^^^^^^^^^^

1 file inspected, 4 offenses detected, 1 offense autocorrectable

1 error occurred:
An error occurred while Style/ParallelAssignment cop was inspecting /tmp/foo.rb:5:4.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2) [x86_64-darwin21]
Finished in 0.25440500024706125 seconds
```

## Steps to reproduce the problem

This ruby file causes the exception to be thrown:
``` ruby
#!/usr/bin/env ruby

class Foo
  def self.init
    @@cvar_one, @@cvar_two = 1, 2
  end
end
```

## RuboCop version

```
1.36.0 (using Parser 3.1.2.1, rubocop-ast 1.21.0, running on ruby 3.1.2) [x86_64-darwin21]
```

**Comment by jimm** [comment]
@koic Thank you for the quick fix!
